tests/TileDB.CSharp.Test/DeletesTest.cs

```diff
@@ -0,0 +1,928 @@
+                { {"rows", new int[16]}, {"cols", new int[16]}, {"a1", new int[16]} };
+            TestUtil.ReadArray(array, layout, readBuffers, ctx: ctx);
+            var expectedData = new Dictionary<string, dynamic>()
+            {
+                { "a1", Enumerable.Range(1, 16).ToArray() },
+                { "rows", new[] { 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4 } },
+                { "cols", new[] { 1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4 } },
+            };
+            TestUtil.CompareBuffers(expectedData, readBuffers, duplicates);
+
+        }
+
+        private void DeleteFour(Context ctx, Array array, LayoutType layout, bool duplicates)
+        {
+            // Delete multiple cells at once
+            array.Open(QueryType.Delete);
+            var deleteQuery = new Query(ctx, array, QueryType.Delete);
+            var queryCondition = new QueryCondition(ctx);
+            queryCondition.Init("cols", 2, QueryConditionOperatorType.Equal);
+            deleteQuery.SetCondition(queryCondition);
+            deleteQuery.Submit();
+            Assert.AreEqual(deleteQuery.Status(), QueryStatus.Completed);
+            Console.WriteLine($"Delete multiple cells status: {deleteQuery.Status()}");
+            array.Close();
+
+            // Check for expected values
+            var readBuffers = new Dictionary<string, dynamic>()
+                { { "rows", new int[16] }, { "cols", new int[16] }, { "a1", new int[16] } };
+            var readQuery = TestUtil.ReadArray(array, layout, readBuffers, ctx: ctx);
+            var expectedData = new Dictionary<string, dynamic>()
+            {
+                { "a1", new[] { 1, 3, 4, 5, 7, 8, 9, 11, 12, 13, 15, 16, 0, 0, 0, 0 } },
+                { "rows", new[] { 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 0, 0, 0, 0 } },
+                { "cols", new[] { 1, 3, 4, 1, 3, 4, 1, 3, 4, 1, 3, 4, 0, 0, 0, 0 } },
+            };
+
+            array.Open(QueryType.Read);
+            Assert.AreEqual(12UL, readQuery.ResultBufferElements()["a1"].Item1);
+            array.Close();
+            TestUtil.CompareBuffers(expectedData, readBuffers, duplicates);
+        }
+
+        // Params for all tests are (layout, legacy, duplicates)
+        [DataRow(LayoutType.Unordered, true, true)]
+        [DataRow(LayoutType.Unordered, true, false)]
+        [DataRow(LayoutType.Unordered, false, true)]
+        [DataRow(LayoutType.Unordered, false, false)]
+        [DataRow(LayoutType.RowMajor, true, true)]
+        [DataRow(LayoutType.RowMajor, true, false)]
+        [DataRow(LayoutType.RowMajor, false, true)]
+        [DataRow(LayoutType.RowMajor, false, false)]
+        [DataRow(LayoutType.GlobalOrder, true, true)]
+        [DataRow(LayoutType.GlobalOrder, true, false)]
+        [DataRow(LayoutType.GlobalOrder, false, true)]
+        [DataRow(LayoutType.GlobalOrder, false, false)]
+        [DataTestMethod]
+        public void MultipleCells(LayoutType layout, bool legacy, bool duplicates)
```

Comment:
Can you replace the `bool legacy` with `string queryReader`, taking `"legacy"` or `"refactored"`? After #157 it will be easier to switch to an enum.